### PR TITLE
ares-cli: init at 3.1.1

### PR DESCRIPTION
**Pull request by Rucadi**
## Description of changes

Ares is the main cli tool used by developers for interacting with LG WebOs Tv, documentation of this tool can be found here:  https://webostv.developer.lge.com/develop/tools/cli-introduction

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Misterio77 — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

Looks good overall. Some feedback:

- Could you format it with `nixfmt-rfc-style`?
- ~As this package is not in repology, it would be nice to have a [`updateScript`](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript) set up.~ See next comment.
  - ~You can probably steal this from other `mkYarnPackage` drvs: [micropad](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/micropad/update.sh), [pomerium](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/http/pomerium/updater.sh), etc.~

File: pkgs/by-name/ar/ares-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  version = "3.0.6";
+  src = fetchFromGitHub {
+    owner = "webos-tools";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-HhS49jg6PrWzXEX5P+NgcznMZw4yJyI0EYoCZ8y6UeU=";
+  };
+
+  yarnLock = ./yarn.lock;
+  doDist = false;
```
Comment (Misterio77) [inline pkgs/by-name/ar/ares-cli/package.nix:16]:
```suggestion
  postPatch = ''
    ln -s npm-shrinkwrap.json package-lock.json
  '';

  dontNpmBuild = true;
  npmDepsHash = "sha256-fTEYQY0ZhwYKnc7+EsgnnGovYJ8w+norq7oLnVrWNZg=";
```

File: pkgs/by-name/ar/ares-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, mkYarnPackage
```
Comment (Misterio77) [inline pkgs/by-name/ar/ares-cli/package.nix:2]:
```suggestion
, buildNpmPackage
```

File: pkgs/by-name/ar/ares-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+, fetchFromGitHub
+}:
+mkYarnPackage rec {
```
Comment (Misterio77) [inline pkgs/by-name/ar/ares-cli/package.nix:5]:
```suggestion
buildNpmPackage rec {
```

**Review by Misterio77 — CHANGES_REQUESTED** [review]
Actually, sorry, one more (bigger) thing.

This is incurring IFD, I believe (that's why ofborg is failing). If you want to go through with `mkYarnPackage`, I believe you'll have to use `fetchYarnDeps` instead of `yarnLock` directly.

But I noticed upstream has a shrinkwrap, you should use `buildNpmPackage` with it instead of generating and vendoring a yarn lock. Here's how to.

PS: for `buildNpmPackage`, `nix-update-script` just works(tm). Simply add this:
```
  passthru.updateScript = nix-update-script { };
```

File: pkgs/by-name/ar/ares-cli/package.nix (hunk 1 above)
Comment (Misterio77) [reply pkgs/by-name/ar/ares-cli/package.nix:16]: `buildNpmPackage` works nicely with `npm-shrinkwrap`, just symlink it to `package-lock.json`.

**Comment by Rucadi** [comment]
> Hey, thanks for the PR!
> 
> Looks good overall. Some feedback:
> 
> * Could you format it with `nixfmt-rfc-style`?
> * ~As this package is not in repology, it would be nice to have a [`updateScript`](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript) set up.~ See next comment.
>   
>   * ~You can probably steal this from other `mkYarnPackage` drvs: [micropad](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/micropad/update.sh), [pomerium](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/http/pomerium/updater.sh), etc.~

Thank you very much for the review, to be honest I didn't quite get why this was failing, since using mkYarnPackage seemed the right way to do it.

I'll try to update the whole PR with your inputs :)

**Comment by Rucadi** [comment]
I pushed the new version and added you as a co-author, your feedback was very appreciated.

It seems like you left the package in a much better way :) However, I wonder if the mkYarnPackage should be fixed? For me it seems like it shouldn't IFD.

Don't hesitate to give additional feedback if needed 😊

**Comment by Misterio77** [comment]
> However, I wonder if the mkYarnPackage should be fixed?

Hmm I think it uses IFD by default for convenience. So that people using it out of nixpkgs tree don't have to bump the deps hash all the time. By using `offlineCache = fetchYarnDeps`, you avoid IFD at the cost of it being a FOD (thus needing a hash).

> Your feedback was very appreciated

I'm glad I could help! :heart:

**Review by Misterio77 — APPROVED** [review]
LGTM :)

File: pkgs/by-name/ar/ares-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    description = "A collection of commands used for creating, packaging, installing, and launching web apps for LG webOS TV.";
+    longDescription = ''
+      webOS CLI (Command Line Interface) provides a collection of commands used for creating, packaging, installing,
+      and launching web apps in the command line environment. The CLI allows you to develop and test your app without using
+      a specific IDE.
+    '';
+    mainProgram = "ares";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ rucadi ];
+    platforms = lib.platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ar/ares-cli/package.nix:37]:
```suggestion
```
buildNpmPackage limits that already to a smaller list of platforms

**Review by JohnRTitor — APPROVED** [review]
I have updated the package to 3.1.1
LGTM, builds for me

**Pull request merged**